### ISSUE TITLE
New update breaks `imp -s` on Ubuntu 22.04

### ISSUE THREAD

**Issue by Xenfo** [arkane-systems/bottle-imp#28]
Update to new `bottle-imp` version, and run `imp -s`. `bottle-imp` was the only updated package since WSL was last working.

![image](https://user-images.githubusercontent.com/46873232/199118769-9f2e2c31-5036-412a-a6cf-cfd8732fbcbf.png)
![image](https://user-images.githubusercontent.com/46873232/199118788-02181cb8-a901-425a-a247-4f7c1f2dc645.png)

While waiting for a patch is there any way to revert the update? I noticed that previous releases are replaced with new release when running `apt update`.

**Comment by cerebrate** [comment]
That's an odd error. First, can you let me see your environment in the host PowerShell (`dir env:`)?

(And, yeah, apt repos are annoying that way. You can always download any previous release from the GitHub releases page, then manually install it with `dpkg --install package_file.deb`.)

**Comment by Xenfo** [comment]
```
PS C:\Users\Samuel Corsi-House> dir env:

Name                           Value
----                           -----
ALLUSERSPROFILE                C:\ProgramData
APPDATA                        C:\Users\Samuel Corsi-House\AppData\Roaming
ChocolateyInstall              C:\ProgramData\chocolatey
ChocolateyLastPathUpdate       133017309726128208
CommonProgramFiles             C:\Program Files\Common Files
CommonProgramFiles(x86)        C:\Program Files (x86)\Common Files
CommonProgramW6432             C:\Program Files\Common Files
COMPUTERNAME                   DESKTOP-RVHOL00
ComSpec                        C:\WINDOWS\system32\cmd.exe
DriverData                     C:\Windows\System32\Drivers\DriverData
FPS_BROWSER_APP_PROFILE_STRING Internet Explorer
FPS_BROWSER_USER_PROFILE_ST... Default
HOMEDRIVE                      C:
HOMEPATH                       \Users\Samuel Corsi-House
LOCALAPPDATA                   C:\Users\Samuel Corsi-House\AppData\Local
LOGONSERVER                    \\DESKTOP-RVHOL00
NUMBER_OF_PROCESSORS           20
OneDrive                       C:\Users\Samuel Corsi-House\OneDrive
OS                             Windows_NT
Path                           C:\Python310\Scripts\;C:\Python310\;C:\Program Files (x86)\Common Files\Oracle\Java\j...
PATHEXT                        .COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW;.CPL
PROCESSOR_ARCHITECTURE         AMD64
PROCESSOR_IDENTIFIER           Intel64 Family 6 Model 151 Stepping 2, GenuineIntel
PROCESSOR_LEVEL                6
PROCESSOR_REVISION             9702
ProgramData                    C:\ProgramData
ProgramFiles                   C:\Program Files
ProgramFiles(x86)              C:\Program Files (x86)
ProgramW6432                   C:\Program Files
PSModulePath                   C:\Users\Samuel Corsi-House\Documents\WindowsPowerShell\Modules;C:\Program Files\Wind...
PUBLIC                         C:\Users\Public
SystemDrive                    C:
SystemRoot                     C:\WINDOWS
TEMP                           C:\Users\SAMUEL~1\AppData\Local\Temp
TMP                            C:\Users\SAMUEL~1\AppData\Local\Temp
USERDOMAIN                     DESKTOP-RVHOL00
USERDOMAIN_ROAMINGPROFILE      DESKTOP-RVHOL00
USERNAME                       Samuel Corsi-House
USERPROFILE                    C:\Users\Samuel Corsi-House
VBOX_MSI_INSTALL_PATH          C:\Program Files\Oracle\VirtualBox\
windir                         C:\WINDOWS
WSLENV                         WT_SESSION::WT_PROFILE_ID
WT_PROFILE_ID                  {61c54bbd-c2c6-5271-96e7-009a87ff44bf}
WT_SESSION                     a45baf8e-202b-4c8a-ad1a-9ee463492d6a
```

**Comment by Xenfo** [comment]
Just used `dpkg` to downgrade back to 0.10, this fixed the issue so it is the new release.

**Comment by cerebrate** [comment]
Yeah, the problem's with the new feature that forwards the WT_SESSION and WT_PROFILE_ID environment variables to the proper login session. Unfortunately, even beyond my not being able to repro it locally, there's really very little _to_ that feature; basically, instead of running

`machinectl shell -q <username>@.host`

as the working part of _imp -s_, it runs

`machinectl shell -E WT_SESSION -E WT_PROFILE_ID shell -q <username>@.host`

and the rest is supposed to be up to _systemd-machined_.

Can you try running those manually inside a WSL session (not yet imp'd)? I'd expect the former to work and the latter to fail with the above error, but I'd like to be sure that's where the issue is.

Also, what version of the _systemd-container_ package do you have installed? I'm wondering if this might be a systemd version issue.

**Comment by Xenfo** [comment]
I get errors on both commands.

`systemd-container/jammy-updates,now 249.11-0ubuntu3.6 amd64 [installed]`

![image](https://user-images.githubusercontent.com/46873232/199133547-3541d7c0-59e4-4095-90c4-e26e24ecd2b4.png)

**Comment by cerebrate** [comment]
Ah, sorry. Forgot to include the `sudo`. Still, that error does clear up what I needed to know, so. Thanks.

**Comment by cerebrate** [comment]
Found it. Apparently the behavior of -E I was relying on was introduced in systemd 250. Should be easy enough to patch, though, so expect 0.12 with the fix tomorrow.

**Comment by cerebrate** [comment]
Fixed in 0.13.
